sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -712,7 +712,9 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
       case r: logical.Range =>
         execution.RangeExec(r) :: Nil
       case r: logical.RepartitionByExpression =>
-        val shuffleOrigin = if (r.optNumPartitions.isEmpty) {
+        val shuffleOrigin = if (r.repartitionWithoutColAndNum) {
```

Comment:
We only call it once. Let's just write `partitionExpressions.isEmpty && optNumPartitions.isEmpty` here

Reply:
OK